### ISSUE TITLE
Screen readers do not announce checked state of ToolStripButton

### ISSUE THREAD

**Issue by dmitrii-drobotov** [dotnet/winforms#8961]
### .NET version

.NET 8

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

Similar to https://github.com/dotnet/winforms/issues/8307, we need to implement [Toggle UIA pattern](https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-implementingtoggle) for `ToolStripButton`, which would allow screen readers to announce the checked state.

#### Current Behavior

For a ToolStrip with following buttons:
* toolStripButton1 with `CheckState = CheckState.Checked` 
* toolStripButton2 with `CheckState = CheckState.Unchecked` and `CheckOnClick = true`
* toolStripButton3 with `CheckState = CheckState.Indeterminate`

Narrator reads only button names, without checked state:
> toolStripButton1, button, 
> toolStripButton2, button, 
> toolStripButton3, button, 

#### Expected Behavior

Screen readers should announce button's checked state:
> toolStripButton1, button, on, 
> toolStripButton2, button, off, 
> toolStripButton3, button, indeterminate,

### Steps to reproduce

1. Add `ToolStripButton`-s to a `ToolStrip` and make buttons checkable: set `CheckOnClick` property to true or `CheckState` property to `Checked`
2. Start Narrator and run the app
3. Focus on tool strip, select checked button — Narrator will not announce that it's checked

Sample app: [ToolStripButtonsTest.zip](https://github.com/dotnet/winforms/files/11190660/ToolStripButtonsTest.zip)

### Additional info

For `PropertyGridToolStripButton`, which extends from `ToolStripButton`, Toggle pattern should not be implemented, because it has "radio button" control type and cannot be toggled on or off.

**Comment by merriemcgaw** [comment]
Thanks for filing this @dmitrii-drobotov!
@rperez030 FYI, this is the .NET 8 bug. We will also be adding checked/unchecked ToolStrip and MenuStrip items to our test application so we are confident in future test passes.

**Comment by dmitrii-drobotov** [comment]
> We will also be adding checked/unchecked ToolStrip and MenuStrip items to our test application so we are confident in future test passes.

Created https://github.com/dotnet/winforms/issues/9005 to implement it.

**Comment by Philip-Wang01** [comment]
Verified this issue in the latest .NET 8.0 Build: 8.0.100-preview4.23330.11, it was fixed that screen readers announce toggle state.
Narrator:
![testnarrator](https://user-images.githubusercontent.com/86937911/233576364-2259a2cb-16a8-4a8b-bfa6-cdb9a802209e.gif)

NVDA:
![testnvda](https://user-images.githubusercontent.com/86937911/233576342-10453d15-e94c-41c2-a9d3-8b66ffec3369.gif)

**Comment by MelonWang1** [comment]
Verified it in VS insertion PR with .Net 8 Preview 4 test pass build: 8.0.0-preview.4.23253.2, issue was fixed. Same test result as above.
